train-2/proportion_thresh_test.cpp

```diff
@@ -0,0 +1,31 @@
+#include<opencv2/opencv.hpp>
+#include<iostream>
+#include<algorithm>
+#include<vector>
+using namespace cv;
+
+void proportion_thresh(const cv::Mat& in, cv::Mat& out,double max_val,double proportion){
+	std::vector<int> vec;
```

Comment:
“Always try `std::vector<...>::reserve(...)` when you know how much you need it...”